src/redux/books/booksSlice.js

```diff
@@ -0,0 +1,56 @@
+import { createSlice } from '@reduxjs/toolkit';
+
+const bookSlice = createSlice({
+  name: 'books',
+  initialState: {
+    books: [],
+    filteredBooks: [],
+    categories: ['Fiction', 'Non-Fiction', 'Science', 'Fantasy'], // Hardcoded list of categories
+    selectedCategory: null,
+  },
+  reducers: {
+    addBook: (state, action) => {
+      state.books.push(action.payload);
+      if (
+        state.selectedCategory === null
+        || state.selectedCategory === action.payload.category
+      ) {
+        state.filteredBooks.push(action.payload);
+      }
+    },
+    removeBook: (state, action) => {
+      const bookIndex = state.books.findIndex(
+        (book) => book.id === action.payload,
+      );
+      if (bookIndex !== -1) {
+        const bookToRemove = state.books[bookIndex];
+        state.books.splice(bookIndex, 1);
+        if (
+          state.selectedCategory === null
+          || state.selectedCategory === bookToRemove.category
+        ) {
+          const filteredBookIndex = state.filteredBooks.findIndex(
+            (book) => book.id === action.payload,
+          );
+          if (filteredBookIndex !== -1) {
+            state.filteredBooks.splice(filteredBookIndex, 1);
+          }
```

Comment:
- [OPTIONAL] I will recommend you do it in a simpler form because reducers are recommended not to have complex argument, it can be as the following:
```
 removeBook: (state, action) => {
      state.books = state.books.filter((book) => book.id !== action.id)
    },
```